CanDIG/katsu
# DIG-1582: Implement overview thresholds & DIG-1533: Change cancer_type to primary_site

## What's new
- Introduce a threshold mechanism to display masked values for small counts, e.g., "<5".
- Rewrite queries for masking and improved performance.
- Update return types from int to str to handle masked values.
- Update schema return structure to accommodate new queries.
- Rename `discover_cancer_type_count` to `discover_primary_site_count`
- Use lazyFunction for enhanced random seeds in testing
- Add a test suite to ensure proper functionality of overview masking for small counts.

### Types of Change(s)

Which other services does it affect?

### Has it been tested for:
- [x] Docker tested

### How to test
- For local, run tox
- For docker, pull this branch, add to docker-compose ` AGGREGATE_COUNT_THRESHOLD=${AGGREGATE_COUNT_THRESHOLD}` and run test-integration
- Optional: Add some data and call overview endpoints to see if value <5 gets properly masked.
